planner/core/rule_eliminate_projection.go

```diff
@@ -109,6 +119,24 @@ func eliminatePhysicalProjection(p PhysicalPlan) PhysicalPlan {
 type projectionEliminator struct {
 }
 
+// ReplaceColumnOfExprInPhysicalProjection replaces column of expression by another PhysicalProjection.
+func ReplaceColumnOfExprInPhysicalProjection(expr expression.Expression, proj *PhysicalProjection, schema *expression.Schema) expression.Expression {
```

Comment:
Should we invoke plan.ResolveIndices() for safety after replacing columns? @winoros

Reply:
It's ok that the `ResolveIndices` happens later than this.

Reply:
I think we can directly move the `ResolveIndices` after the current post optimize phase.